ui/app/src/components/workspaces/WorkspaceProvider.tsx

```diff
@@ -26,46 +28,75 @@ export const WorkspaceProvider: React.FunctionComponent = () => {
-        setWorkspaceServices(workspaceServices.workspaceServices);
-        setLoadingState(wsRoles && wsRoles.length > 0 ? LoadingState.Ok : LoadingState.AccessDenied);
-
-        // get shared services to pass to nav shared services pages
-        const sharedServices = await apiCall(ApiEndpoint.SharedServices, HttpMethod.Get);
-        setSharedServices(sharedServices.sharedServices);
-
-      } catch (e: any){
-        e.userMessage = 'Error retrieving workspace';
-        setApiError(e);
-        setLoadingState(LoadingState.Error);
+        let ws: Workspace = {} as Workspace;
+
+        if (authProvisioned) {
+          try {
+            ws = (await apiCall(`${ApiEndpoint.Workspaces}/${workspaceId}`, HttpMethod.Get, scopeId)).workspace;
+            workspaceCtx.current.setWorkspace(ws);
+            // use the client ID to get a token against the workspace (tokenOnly), and set the workspace roles in the context
+
+            await apiCall(`${ApiEndpoint.Workspaces}/${workspaceId}`, HttpMethod.Get, ws.properties.scope_id, undefined, ResultType.JSON, (roles: Array<string>) => {
+              workspaceCtx.current.setRoles(roles);
+              wsRoles = roles;
+            }, true);
+          } catch (e: any) {
+            // do nothing, we need to try as TRE Admin
+          }
+        }
+
+        if (wsRoles.length > 0) {
+
+          // get workspace services to pass to nav + ws services page
+          const workspaceServices = await apiCall(`${ApiEndpoint.Workspaces}/${ws.id}/${ApiEndpoint.WorkspaceServices}`, HttpMethod.Get, ws.properties.scope_id);
+          setWorkspaceServices(workspaceServices.workspaceServices);
+          setLoadingState(LoadingState.Ok);
+          // get shared services to pass to nav shared services pages
+          const sharedServices = await apiCall(ApiEndpoint.SharedServices, HttpMethod.Get);
+          setSharedServices(sharedServices.sharedServices);
+        } else if (appRoles.roles.includes(RoleName.TREAdmin)) {
+
+          ws = (await apiCall(`${ApiEndpoint.Workspaces}/${workspaceId}`, HttpMethod.Get)).workspace;
+          workspaceCtx.current.setWorkspace(ws);
+          setLoadingState(LoadingState.Ok);
+          refIsTREAdminUser.current = true;
+        } else {
+          let e = new APIError();
+          e.status = 403;
+          e.userMessage = "User does not have a role assigned in the workspace or the TRE Admin role assigned";
+          e.endpoint = `${ApiEndpoint.Workspaces}/${workspaceId}`;
+          throw e;
+        }
+
+      } catch (e: any) {
+        if (e.status === 401 || e.status === 403) {
+          setApiError(e);
+          setLoadingState(LoadingState.AccessDenied)
+        } else {
+          e.userMessage = 'Error retrieving workspace';
+          setApiError(e);
+          setLoadingState(LoadingState.Error);
+        }
       }
+
+      console.log('loading state: ' + loadingState);
```

Comment:
Do we want to log this? Already a lot of noise in the logger for TRE. If want to keep for debug purposes, use console.debug() which will only shows these logs when debug is enabled in the web dev console

Reply:
Gone, was me debugging.